galaxy/api/views/search.py

```diff
@@ -236,16 +237,17 @@ def add_namespaces_filter(queryset, namespaces):
     def add_platforms_filter(queryset, platforms):
         if not platforms:
             return queryset
-        return queryset.filter(
-            platforms__in=models.Platform.objects.filter(name__in=platforms))
+        platforms_qs = models.Content.objects.only('pk').filter(
+            platforms__name__in=platforms)
+        return queryset.filter(pk__in=platforms_qs)
```

Comment:
Is writing it this appreciably different in the resulting SQL from

```return queryset.filter(platforms__name__in=platforms)```

Reply:
This call ```queryset.filter(platforms__name__in=platforms)``` generates SQL JOIN query, for example:

```
SELECT ...
FROM main_content
JOIN main_content_platforms ON ...
JOIN main_platforms ON ...
WHERE main_platforms.name IN (...)
```
Such JOIN query on many-to-many relationship creates row duplicates, that then can be eliminated by DISTINCT \ DISTINCT ON. Applying dumb `.distinct()` call on a queryset leads to generating `SELECT DISTINCT` over all fields in query, which for this select query can be large (considering selecting fields from related tables).

One possible solution is to select records and join with subquery that returns unique content ids, but Django ORM doesn't provide an obvious interface for custom joins.

Another solution, we use in search, is making `WHERE 'id' IN (SELECT ...)` query, which has similar performance to JOIN with unique IDs.